src/datasets/iterable_dataset.py

```diff
@@ -1350,9 +1351,11 @@ def _concatenate_iterable_datasets(
     else:
         _check_column_names([col_name for dset in dsets for col_name in dset.features])
 
-    features = Features()
-    for dset in dsets:
-        features.update(dset.features)
+    # TODO: improve this to account for a mix of ClassLabel and Value for example
+    # right now it would keep the type of the first dataset in the list
+    features = Features(
+        {k: v for features in _align_features([dset.features for dset in dsets[::-1]]) for k, v in features.items()}
+    )
```

Comment:
I believe `[::-1]` is not needed here since we use `_check_if_features_can_be_aligned` + `_align_features`:
```suggestion
    features = Features(
        {k: v for features in _align_features([dset.features for dset in dsets]) for k, v in features.items()}
    )
```